pins/boards.py

```diff
@@ -915,6 +919,20 @@ def pin_write(self, *args, access_type=None, **kwargs):
                 " can write to it."
             )
 
+        n_versions_before = 0
+        if self.pin_exists(pin_name):
+            versions_df = self.pin_versions(pin_name, as_df=True)
+            versions = versions_df["version"].to_list()
+            n_versions_before = len(versions)
+
+        if versioned is None:
+            versioned = True if n_versions_before > 1 else self.versioned
+        if versioned is False and n_versions_before > 1:
+            raise PinsVersionError(
```

Comment:
trying to fail fast if we can't write the pin anyway

Reply:
It seems like the effects of this are...

* upside: it will take less time on connect to raise a PinsVersionError.
* downside: every successful use of pin_write will be slower.

I think it's okay for failing cases to run slightly slower, since they should be the exception, not the norm. I would remove, so the normal cases don't eat the overhead of running these calls multiple times. (But also might be misunderstanding this code).

Reply:
Updated-- then was reminded the Posit Connect cannot delete the latest pin 😅 so there is a need to do these calls *before* a pin is written